add nodejs corepack page (#8)
navarroaxel/navarroaxel.github.io: blog/nodejs-corepack.md

> # Corepack the Node.js' manager of package managers

Reviewer: ```suggestion
# Corepack: the Node.js' manager of package managers
```

> Corepack is the new built-in tool to prepare the required package manager for our projects. We're going to see what comes with this new package for packagers. 📦

Reviewer: ```suggestion
Corepack is the new built-in tool for preparing the required package manager for our projects. We're going to see what comes with this new package for packagers. 📦
```

> Corepack is the new built-in tool for preparing the required package manager for our projects. We're going to see what comes with this new package for packagers. 📦

Reviewer 2: > for preparing the required package manager for our projects

why is it required? maybe a little context on this would help on clarity

> Corepack is the new built-in tool for preparing the required package manager for our projects. We're going to see what comes with this new package for packagers. 📦

Reviewer 3: ```suggestion
Corepack is the new built-in tool for preparing the required package manager for our projects. Let's see what comes with this new package for packagers. 📦
```

## Installation

> We're in track to get the new Node.js LTS v16 at the end of October, without a fancy name assigned yet, and we'll get [Corepack](https://github.com/nodejs/corepack) preinstalled in the default Node.js configuration since the [v16.9.0](https://github.com/nodejs/node/releases/tag/v16.9.0). 👌

Reviewer: ```suggestion
We're in track to get the new Node.js LTS v16 at the end of October, without a fancy name assigned yet, and we're getting [Corepack](https://github.com/nodejs/corepack) preinstalled in the default Node.js configuration since [v16.9.0](https://github.com/nodejs/node/releases/tag/v16.9.0). 👌
```

> We're in track to get the new Node.js LTS v16 at the end of October, without a fancy name assigned yet, and we're getting [Corepack](https://github.com/nodejs/corepack) preinstalled in the default Node.js configuration since [v16.9.0](https://github.com/nodejs/node/releases/tag/v16.9.0). 👌

Reviewer 3: ```suggestion
The new Node.js LTS v16 will be released at the end of October (without a fancy name assigned yet), it'll have [Corepack](https://github.com/nodejs/corepack) preinstalled in the default configuration since [v16.9.0](https://github.com/nodejs/node/releases/tag/v16.9.0). 👌
```

> We're in track to get the new Node.js LTS v16 at the end of October, without a fancy name assigned yet, and we're getting [Corepack](https://github.com/nodejs/corepack) preinstalled in the default Node.js configuration since [v16.9.0](https://github.com/nodejs/node/releases/tag/v16.9.0). 👌

Reviewer 2: > We're in track to get the new Node.js LTS v16 at the end of October, without a fancy name assigned yet

We're on track to a new Node.js version, LTS v16, at the end of October, fancy name TBD.

> If we want this tool for previous versions we just need to install it as a global package:

Reviewer: ```suggestion
If we want to use this tool with previous versions we just need to install it as a global package:
```

```bash
$ npm install -g corepack
```

> 🧠 Remember that until now, the Corepack v0.9.0 only supports `pnpm`, `npm` and `yarn`.

Reviewer: ```suggestion
🧠 Remember that, until now, Corepack v0.9.0 was only supporting `pnpm`, `npm` and `yarn`.
```

> 🧠 Remember that, until now, Corepack v0.9.0 was only supporting `pnpm`, `npm` and `yarn`.

Reviewer 3: ```suggestion
🧠 Remember that for now Corepack v0.9.0 only supports `pnpm`, `npm` and `yarn`.
```

## Basic usage

> We can use use the package manager that we always use in any machine without worry if it is installed or not using by example:

Reviewer: ```suggestion
We can use the same package manager that we always use in any machine without worring if it's installed or not, using:
```

> We can use the same package manager that we always use in any machine without worring if it's installed or not, using:

Reviewer 2: > We can use the same package manager that we always use in any machine without worring if it's installed or not, using:

We can use the same package manager that we've always used in any machine without worrying if it's installed or not, using:

Author: ```suggestion
We can use the same package manager that we've always use in any machine without worrying if it's installed or not, using:
```

```bash
$ corepack pnpm install
```

> Corepack will take care that the required manager is ready to process your request.

Reviewer: ```suggestion
Corepack will take care of that the required manager is ready to process your request.
```

> Corepack will take care of that the required manager is ready to process your request.

Reviewer 3: ```suggestion
Corepack will see to it that the required manager is ready to process your request.
```
I didn't knew [this expression](https://www.merriam-webster.com/dictionary/see%20to%20it).

### Specific versions

> The included `npm` version with Node.js LTS v16 is v7, so what if we need `npm@6` for one project and `npm@7` for other?

Reviewer: ```suggestion
The included `npm` version with Node.js LTS v16 is v7, so what if we need `npm@6` for one project and `npm@7` for another?
```

> The included `npm` version with Node.js LTS v16 is v7, so what if we need `npm@6` for one project and `npm@7` for another?

Reviewer 2: > The included `npm` version with Node.js LTS v16 is v7, so, what if we need `npm@6` for one project and `npm@7` for another? 🤔

> Well, you will find `corepack` very similar to `nvm` in this scenario.

Reviewer: ```suggestion
Well, in this scenario you'll will find `corepack` very similar to [`nvm`](https://github.com/nvm-sh/nvm).
```

```bash
$ corepack prepare npm@6.14.15 --activate
$ node -v
v16.9.0
$ npm -v
7.21.1
$ corepack npm -v
6.14.15
```

> 🤯 So, with `corepack prepare` you'll setup `npm@6`, and you can use it every time you need just prefixing your `npm` usage with Corepack, e.g. `corepack npm ci`.

Reviewer: ```suggestion
🤯 So with `corepack prepare` you'll setup `npm@6`, and you can use it every time you need just by prefixing your `npm` usage with Corepack, e.g. `corepack npm ci`.
```

And your global `npm` command will continue untouched being `npm@7`.

> 🤔 But what is that `--activate` flag? Corepack will keep a cache of used versions of a packager, so you should activate which version you want to use, otherwise you're still using `npm@7` when you use `corepack npm -v`.

Reviewer: ```suggestion
🤔 But what is that `--activate` flag? Corepack will keep a cache of used versions of a packager, so you should activate the specific version that you want to use; otherwise you will still be using `npm@7` when you use `corepack npm -v`.
```

## Global package managers 🌎

What if I want `pnpm` as a global command in my terminal? You can enable Corepack to handle `pnpm` and automatically install it when is used for the first time:

```bash
corepack enable
```

Corepack will install the handlers in the node directory because it's in the `$PATH`, but you can use the destination of your choice:

```bash
corepack enable --install-directory path/to/bins_folder
```

> And, it's done, you can use `pnpm` or `yarn` directly:

Reviewer: ```suggestion
And, it's done you can use `pnpm` or `yarn` directly:
```

> And, it's done you can use `pnpm` or `yarn` directly:

Reviewer 3: ```suggestion
And it's done, you can use `pnpm` or `yarn` directly:
```

```bash
$ pnpm -v
6.11.0
```

## Docker images

> At the moment, the official Node.js images in Docker comes with `npm` and `yarn` preinstalled, but maybe this can change in a near future. (no more `npm` and `yarn` preinstalled in the Docker image it's possible! 🤯)

Reviewer: ```suggestion
At the moment, the official Node.js images in Docker comes with `npm` and `yarn` preinstalled, but this may change in a near future. (no more `npm` and `yarn` preinstalled in the Docker image is possible! 🤯)
```

> At the moment, the official Node.js images in Docker comes with `npm` and `yarn` preinstalled, but this may change in a near future. (no more `npm` and `yarn` preinstalled in the Docker image is possible! 🤯)

Reviewer 3: ```suggestion
At the moment, the official Node.js images in Docker comes with `npm` and `yarn` preinstalled, but this may change in the near future. (no more `npm` and `yarn` preinstalled in the Docker image is possible! 🤯)
```

> Previously, if you use `pnpm` in a Dockerfile you need to install it using `npm` or `curl`:

Reviewer: Which tense are you trying to use here? If you mean present, I suggest you to say _Before using /installing ..._ or sth instead of _Previously_. Otherwise, if you are talking about the past, you should conjugate the verbs in that tense to make the sentence clearer.
Disclaimer: maybe this seems clear for a native. Idk

Author: ```suggestion
If you use `pnpm` in Docker you need to install it using `npm` or `curl`:
```

```bash

> RUN npm install -g pnpm && pnpm install

Reviewer: ```suggestion
$RUN npm install -g pnpm && pnpm install
```

```

Now, Corepack is here to make this easier to you:

```bash

> RUN corepack pnpm install

Reviewer: ```suggestion
$RUN corepack pnpm install
```

Author: The `Dockerfile` is not a REPL (interactive shell)

> Now, Corepack is here to make this easier to you:

Reviewer 3: ```suggestion
Now Corepack is here to make this easier for you:
```

```

> Or, maybe you need a specific version like this:

Reviewer: ```suggestion
Or maybe you need a specific version like this:
```

```bash

> RUN corepack prepare npm@6.14.15 --activate && corepack npm ci

Reviewer: ```suggestion
$RUN corepack prepare npm@6.14.15 --activate && corepack npm ci
```

```

## Conclusion

> With Corepack we can have more control about which package manager we're using and we don't need to check if Node.js upgrades `npm` to v8 and we didn't notice that event. Until we notice that our application is on fire. 🔥

Reviewer: ```suggestion
With Corepack we can have more control about which package manager we're using and we don't need to check if Node.js upgrades `npm` and we don't notice. Because, if that happens, until we realize that our application is on fire. 🔥
```

> With Corepack we can have more control about which package manager we're using and we don't need to check if Node.js upgrades `npm` and we don't notice. Because, if that happens, until we realize that our application is on fire. 🔥

Reviewer 2: Well, because if that happens, by the time we realize that.. our application is on fire. 🔥

> With Corepack we can have more control about which package manager we're using and we don't need to check if Node.js upgrades `npm` and we don't notice. Because, if that happens, until we realize that our application is on fire. 🔥

Reviewer 3: ```suggestion
With Corepack we can have more control about which package manager we're using and we don't need to check if Node.js upgrades `npm` and we don't notice. If that happens we'll realize of it just when our application is on fire. 🔥
```

> Also, `npm` is part of GitHub, Inc. and it's not part of the Node.js governance, it's a good decision if the Node.js project becomes agnostic and as developers can use the package manager that we need without waste space in our Docker images for packagers that we won't use.

Reviewer: ```suggestion
Also, `npm` is part of GitHub, Inc. and it's not part of the Node.js governance, so it's a good decision if the Node.js project becomes agnostic and as developers we can use the package manager that we need without wasting space in our Docker images for packagers that we won't use.
```

> Also, `npm` is part of GitHub, Inc. and it's not part of the Node.js governance, so it's a good decision if the Node.js project becomes agnostic and as developers we can use the package manager that we need without wasting space in our Docker images for packagers that we won't use.

Reviewer 3: ```suggestion
Also, `npm` is part of GitHub Inc. and it's not part of the Node.js governance, so it's a good decision if the Node.js project becomes agnostic and as developers we can use the package manager that we need without wasting space in our Docker images on packagers that we won't use.
```
